pipeline_steps/common.groovy

```diff
@@ -439,24 +439,50 @@ def prepareRpcGit(String branch = "auto", String dest = "/opt"){
 
     print("Repo: ${env.RPC_REPO} Branch: ${branch}")
 
-    // checkout used instead of git as a custom refspec is required
-    // to checkout pull requests
-    checkout([$class: 'GitSCM',
-      branches: [[name: branch]],
-      doGenerateSubmoduleConfigurations: false,
-      extensions: [[$class: 'CleanCheckout']],
-      submoduleCfg: [],
-      userRemoteConfigs: [
-        [
-          url: env.RPC_REPO,
-          refspec: '+refs/pull/*:refs/remotes/origin/pr/* +refs/heads/*:refs/remotes/origin/*'
-        ]
-      ]
-    ]) // checkout
-    sh "git submodule update --init"
+    clone_with_pr_refs(env.RPC_REPO, branch)
   } // dir
 }
 
+// Clone repo with Refspecs required for PRs.
+// Should't need to supply any params to checkout a PR merged with the base.
+// Uses shell+git to avoid hostname verification failures with
+// the built in git scm step.
+// Use init + fetch instead of clone so that the repo can
+// be cloned into a non-empty directory. Thats added for
+// compatibility with the jenkins git scm step.
+void clone_with_pr_refs(
+  String repo="https://github.com/${env.ghprbGhRepository}.git",
+  String ref="origin/pr/${env.ghprbPullId}/merge",
+  String refspec='+refs/pull/\\*:refs/remotes/origin/pr/\\*'\
+                +' +refs/heads/\\*:refs/remotes/origin/\\*'
+){
+  if(repo == "https://github.com/.git"){
+    throw new Exception(
+      "repo not supplied to common.clone_with_pr_refs or env.ghprbGhRepository"\
+      + " not set."
+    )
+  }
+  if(ref == "origin/pr//merge"){
+    throw new Exception(
+      "ref not supplied to common.clone_with_pr_refs or env.ghprbPullID not "\
+      + "set, attempting to checkout PR for a periodic build?")
+  }
+  print "Cloning Repo: ${repo}@${ref}"
+  sshagent (credentials:['rpc-jenkins-svc-github-ssh-key']){
+    sh """/bin/bash -xe
+      mkdir -p ~/.ssh
+      ssh-keyscan github.com >> ~/.ssh/known_hosts
+      # use init + fetch to avoid the "dir not empty git fail"
+      git init .
+      git remote add origin "${repo}"
+      # Don't quote refspec as it should be separate args to git.
+      git fetch --tags origin ${refspec}
```

Comment:
Private repos can be cloned via ssh or http as long as its an authenticated connection. I want to support both types of URL, so I've added the ssh agent for ssh urls, and a git credentials store for https urls.